Relay-Net 8 channel Relay and binary input module, initial commit

hvbommel opened the pull request:
This is a new plugin for the Relay-Net-V5.7 relay and binary input card. I don't know the manufacturer but you can find this low lost card for example on ebay or aliexpress. On installation 8 switches and 8 inputs are created. The relays can be activated /de activated by operating the switch. Both the inputs and the status of the relays can optionally be polled at a selected poll interval. Besides polling of the inputs, polling of the relays is usefull. This allows domoticz to always display and act on the actual relay status that could have been modified by some external http tools running in a browser for example. Only one controller, Domoticz in this case, should attempt to connect to the relay card using TCP/IP. On poll intervals faster then once per 10 seconds the TCP connection is left open al the time. For slower poll intervals the connection is initiated/terminated at each poll interval.

hvbommel committed:
Relay-Net 8 channel Relay and binary input module, initial commit

gizmocuz commented:
I think you need to add the .cpp to CMakeList.txt as well (for the linux users), did you test the code under linux?

hvbommel commented:
Hello,

 

Correct.

You should find the .cpp online 363 in CMAkeLists.txt: “hardware/RelayNet.cpp”. If all is well this is the file that I pushed.

 

I have tested on Ubuntu 16.04 and Raspberry Pi 3 model B.

 

Henri

 

Van: gizmocuz [mailto:notifications@github.com] 
Verzonden: dinsdag 21 februari 2017 15:11
Aan: domoticz/domoticz <domoticz@noreply.github.com>
CC: hvbommel <info@hotballoon.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [domoticz/domoticz] Relay-Net 8 channel Relay and binary input module, initial commit (#1248)

 

I think you need to add the .cpp to CMakeList.txt as well (for the linux users), did you test the code under linux?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/domoticz/domoticz/pull/1248#issuecomment-281354459> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AOHR1mIIbctWaWlyXcZrKuU6DipTX1uRks5revB_gaJpZM4MHSC1> .  <http://www.baruwa.org/1x1spacer.gif> 

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/domoticz/domoticz","title":"domoticz/domoticz","subtitle":"GitHub repository","main_image_url":" <https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png> https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":" <https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png> https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":" <https://github.com/domoticz/domoticz> https://github.com/domoticz/domoticz"}},"updates":{"snippets":[{"icon":"PERSON","message":"@gizmocuz in #1248: I think you need to add the .cpp to CMakeList.txt as well (for the linux users), did you test the code under linux?"}],"action":{"name":"View Pull Request","url":" <https://github.com/domoticz/domoticz/pull/1248#issuecomment-281354459> https://github.com/domoticz/domoticz/pull/1248#issuecomment-281354459"}}}

gizmocuz commented:
Okey, then please have a look at the appveyor build log what else is wrong... you also did add it to the visual studio project?
Please do NOT reply by email, or at least do not include a copy of the mail you received

hvbommel commented:
I am currently using eclipse on ubuntu to edit, test and debug the project. I created the eclipse project based on the CMakelists.txt (cmake -G"Eclipse CDT4 - Unix Makefiles" -DCMAKE_BUILD_TYPE=debug CMakeLists.txt). I looked at the log and indeed do see that the "hardware/RelayNet.cpp" is missing. For some reason it looks like CMakeLists.txt was not updated. 

This is my change log from gitk and we can see the "+hardware/RelayNet.cpp" was added.
 hardware/PVOutput_Input.cpp
 hardware/Razberry.cpp
 hardware/Rego6XXSerial.cpp
+hardware/RelayNet.cpp
 hardware/RFLinkBase.cpp
 hardware/RFLinkSerial.cpp
 hardware/RFLinkTCP.cpp

I will try to sort out what is wrong and get back later.

hvbommel committed:
Stage and commit the project files

hvbommel commented:
This is what I see in the gitbub change log for CMakeLists.txt . All is also right on my local system.
 
360    hardware/PVOutput_Input.cpp  
361    hardware/Razberry.cpp  
362    hardware/Rego6XXSerial.cpp  
363   +hardware/RelayNet.cpp  
364    hardware/RFLinkBase.cpp  
365    hardware/RFLinkSerial.cpp  
366    hardware/RFLinkTCP.cpp  

Could it be that the build was run before the project was done using the old CMakeLists.txt ie with the missing RelayNet.cpp entry ? I have staged, committed and pushed the .project files, is that helping ?

hvbommel commented:
New pull request, after adding the .project files

hvbommel commented:
I looks like the build procedure is not generating new project contents using the CMakeLists.txt ie ignoring the CMakeLists.txt. Could that be ?

hvbommel commented:
I could move to Visual Studio if that is the only way out.

gizmocuz commented:
Sorry, it's the Visual Studio project that does not have the files

hvbommel commented:
I will make it a Visual Studio project, build it and also test it on Windows. Then I will push the changes.  It can take a day or two. I need to do it in the evening.

hvbommel committed:
Create Visual studio project

Visual studio project created. It is not compiling because
of this error: The CodeDom provider type
"Microsoft.VisualC.CppCodeProvider, CppCodeProvider, Version=10.0.0.0,
Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" could not be located.
I am doing the push to see if the code now builds on github

hvbommel commented:
After creating the Visual studio project I did an push of the project files to see if it now builds on github. (try on error)

hvbommel committed:
Trying to fix github build

In CMakeLists.txt removed & entered again RelayNet.txt
renamed (ie effectivly romeved) the eclipse project files.
new updated VS project files

hvbommel commented:
This time from windows, a new push with some project file modifications

hvbommel commented:
one more try to see if RelayNet.cpp is compiled or not

hvbommel committed:
Add cpp to msbuild

Opened the solution, added RelayNet.cpp & RelayNet.h

hvbommel commented:
if widnows then sys/socket.h -> winsock2.h

hvbommel commented:
will first make my local windows build work, then push it here again. I take installed a new windows 10, put VS on it and now I am going to put all prerequisites like boost etc on

gizmocuz commented:
Yes, always better to test with a fork ;)
Probably for merging i can make 1 comment ;)
But best to fork en then push
